ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -191,7 +194,20 @@ def create_input_processors(
             embedding_sizes.append(embedding_size)
             if encoder is None:
                 var_len_indices.append(idx)
-        return (nn.ModuleList(encoders), embedding_sizes, var_len_indices)
+
+        x_self_size = sum(embedding_sizes)  # The size of the "self" embedding
+        for idx in var_len_indices:
+            entity_max: int = obs_spec[idx].shape[0]
+            var_encoders.append(
+                EntityEmbedding(x_self_size, obs_spec[idx].shape[1], entity_max, h_size)
+            )
+            entity_num_max_elements += entity_max
+        return (
+            nn.ModuleList(encoders),
+            nn.ModuleList(var_encoders),
+            embedding_sizes,
+            entity_num_max_elements,
```

Comment:
I think it would be cleaner if `entity_num_max_elements` was calculated from `nn.ModuleList(var_encoders)` in network body instead of returned by this util method.